test-runner/src/main/scala/com/typesafe/spark/test/mesos/framework/runners/MultiClusterModeRunner.scala

```diff
@@ -0,0 +1,129 @@
+package com.typesafe.spark.test.mesos.framework.runners
+
+import java.io.File
+import java.net.URL
+
+import Utils._
+import com.typesafe.config.{Config, ConfigFactory}
+import com.typesafe.spark.test.mesos.mesosstate.MesosCluster
+
+import org.apache.zookeeper.{ZKUtil, ZooKeeper, KeeperException}
+
+import scala.sys.process.Process
+
+object MultiClusterModeRunner {
+  def deleteZNodes(zk: ZooKeeper, path: String): Unit = {
+    try {
+      ZKUtil.deleteRecursive(zk, path);
+    } catch {
+      // Ignoring if the node doesn't exist.
+      case e: KeeperException.NoNodeException => {}
+    }
+  }
+
+  def run(args: Array[String])(implicit config: Config): Int = {
+    val sparkHome = args(0)
+    val mesosMasterUrl = args(1)
+    val applicationJarPath = args(2)
+    val mesosConsoleUrl = mesosMasterUrl.replaceAll("mesos://", "http://")
+
+    val hostAddress = if (config.hasPath("docker.host.ip")) config.getString("docker.host.ip")
+      else "localhost"
+
+    val jarPath = if (config.hasPath("hdfs.uri")) {
+      val hdfsUri = config.getString("hdfs.uri")
+      // Copying application jar to docker location so mesos slaves can pick it up.
+      val hdfsJarLocation = Utils.copyApplicationJar(args(2), hdfsUri)
+      printMsg(s"Application jar file is copied to HDFS $hdfsJarLocation")
+      hdfsJarLocation
+    } else {
+      applicationJarPath
+    }
+
+    // Make sure we kill any running mesos frameworks. Right now if we run
+    // mesos dispatcher it doesn't die automatically.
+    killAnyRunningFrameworks(mesosConsoleUrl)
+
+    val zookeeperDir = config.getString("mesos.dispatcher.zookeeper.dir")
+    val zkConnection = config.getString("spark.zk.uri")
```

Comment:
config.getString("spark.zk.uri").replaceAll("zk://", "") otherwise it fails with:

java.lang.IllegalArgumentException: Invalid path string "//172.17.0.1:2181" caused by empty node name specified @1

cat test-runner/mit-application.conf 
spark.zk.uri = "zk://172.17.0.1:2181"